carousel/core/simulations.py

```diff
@@ -450,22 +453,23 @@ def start(self, model, progress_hook=None):
                                 data, limits in self.thresholds.iteritems())
             else:
                 night = None
-            # TODO: maybe add ``start_at`` parameter combined with ``frequency``
-            # Determine if calculation is scheduled for this timestep
-            if not calc_reg.frequency[calc].dimensionality:
-                is_scheduled = (idx_tot % calc_reg.frequency[calc]) == 0
-            else:
-                # Frequency with units of time
-                is_scheduled = (
-                    (idx_tot * self.interval % calc_reg.frequency[calc]) == 0
-                )
             # daytime or always calculated outputs
             for calc in self.calc_order:
-                if (not night or calc_reg.always_calc[calc]) and is_scheduled:
-
-                    calc_reg[calc].calc_dynamic(
-                        idx, formula_reg, data_reg, out_reg,
-                        timestep=self.interval
+                # Determine if calculation is scheduled for this timestep
+                # TODO: add ``start_at`` parameter combined with ``frequency``
+                freq = calc_reg.frequency[calc]
+                if not freq.dimensionality:
+                    is_scheduled = (idx_tot % freq) == 0
+                else:
+                    # Frequency with units of time
+                    is_scheduled = ((idx_tot * self.interval) % freq) == 0
+                is_scheduled = (
```

Comment:
This was specifically implemented for [`solder_joint_fail`](../../PVSimLife/blob/master/pvsimlife/degmodes/solder_joint_fail.json) which has a daily calculation frequency, no matter what the timestep is.